### PR TITLE
Support list of masks in AugmentationSequential

### PR DESCRIPTION
**Pull request by ashnair1**
#### Changes

First draft for supporting list of masks

Fixes https://github.com/kornia/kornia/discussions/2417

@edgarriba @shijianjian @johnnv1 

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/augmentation/container/ops.py [hunk 1]
```diff
@@ -195,32 +197,85 @@ def transform(cls, input: Tensor, module: Module, param: ParamItem, extra_args:
...
         elif isinstance(module, (_AugmentationBase)):
-            input = module.transform_masks(
-                input, params=cls.get_instance_module_param(param), flags=module.flags, **extra_args
-            )
+            if isinstance(input, list):
+                tfm_input = []
+                for inp in input:
+                    inp = module.transform_tensor(inp)
+                    tfm_inp = module.transform_masks(
+                        inp, params=cls.get_instance_module_param(param), flags=module.flags, **extra_args
```
Comment (shijianjian) [inline kornia/augmentation/container/ops.py:234]: The params are shared across different masks in the list I guess. So, maybe it is better to retrieve the params only once.

**Review by shijianjian — COMMENTED** [review]
LGTM. Can you add some tests? Probably update the tutorials to say that we are supporting instance segmentation now.

File: kornia/augmentation/container/ops.py (hunk 1 above)
Comment (ashnair1) [reply kornia/augmentation/container/ops.py]: Done

File: kornia/augmentation/container/ops.py [hunk 2]
```diff
@@ -185,42 +185,91 @@ class MaskSequentialOps(SequentialOpsInterface[Tensor]):
...
         if isinstance(module, (K.GeometricAugmentationBase2D,)):
-            input = module.transform_masks(
-                input,
-                params=cls.get_instance_module_param(param),
-                flags=module.flags,
-                transform=module.transform_matrix,
-                **extra_args,
-            )
+            if isinstance(input, list):
+                tfm_input = []
```
Comment (edgarriba) [inline kornia/augmentation/container/ops.py:201]: can you somehow factor this functionality into a single call `transform_masks_list` or similar name

Comment (johnnv1) [reply kornia/augmentation/container/ops.py:201]: +1

**Review by johnnv1 — APPROVED** [review]
LGTM, just add some tests

File: kornia/augmentation/container/ops.py [hunk 3]
```diff
@@ -223,6 +226,65 @@ def transform(cls, input: Tensor, module: Module, param: ParamItem, extra_args:
...
+        """Apply a transformation with respect to the parameters.
+
+        Args:
+            input: list of input tensors.
+            module: any torch Module but only kornia augmentation modules will count
+                to apply transformations.
+            param: the corresponding parameters to the module.
+        """
+        if isinstance(module, (K.GeometricAugmentationBase2D,)):
+            tfm_input = []
```
Comment (edgarriba) [inline kornia/augmentation/container/ops.py:242]: What I meant by factoring was to remove boilerplate code of the next four blocks that are practically the same code and put into a single function. Can be done in a subsequent PR

**Review by edgarriba — APPROVED** [review]

**Comment by shijianjian** [comment]
Hi @ashnair1, Can you add some tests under test/augmentation/test_sequential?

**Comment by shijianjian** [comment]
Hi @ashnair1, we are targeting at a release in this weekend. It is better if we can include this PR in the release.

**Comment by ashnair1** [comment]
Will try to add tests soon

**Comment by shijianjian** [comment]
Hi @ashnair1, any updates on this?

**Comment by johnnv1** [comment]
hey @ashnair1, will you be able to add tests for it?

**Comment by ashnair1** [comment]
Not in the short term, unfortunately.

**Comment by johnnv1** [comment]
hey @ashnair1 i have pushed your chances on #2740, and will try to add the tests to finish this patch. if you want to help, you're very welcome 😄. In any case, I will be closing this one, in favor of #2740, which is aligned with the main branch.

**Pull request closed without merging**